sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -147,7 +147,7 @@ public virtual async Task<Response<ConfigurationSetting>> AddConfigurationSettin
                     case 201:
                         return await CreateResponseAsync(response, cancellationToken).ConfigureAwait(false);
                     case 412:
-                        throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response, "Setting was already present.").ConfigureAwait(false);
+                        throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response, new ResponseError(null, "Setting was already present.")).ConfigureAwait(false);
```

Comment:
Do we care about an extra allocation here?  Exception machinery means we're already not in a high-performance scenario?

Reply:
This isn't an extra allocation, if we were not allocating here, we would allocate when trying to parse the response. But in general, the cost of this allocation is very small in comparison to throwing and processing the exception.